src/lib/dashboard/utils.ts

```diff
@@ -0,0 +1,316 @@
+    );
+  }
+
+  const currentTime = new Date();
+  const targetTime = new Date(currentTime.getTime());
+
+  if (unit === "h") {
+    targetTime.setHours(currentTime.getHours() - value);
+  } else {
+    targetTime.setDate(currentTime.getDate() - value);
+  }
+
+  return targetTime.toISOString().split(".")[0] + "Z";
+}
+
+export function formatDateTime(dateString: string): string {
+  const date = new Date(dateString);
+
+  const day = date.getDate();
+  const month = new Intl.DateTimeFormat("en-US", { month: "short" }).format(
+    date
+  );
+  const year = date.getFullYear();
+  const hours = date.getHours();
+  const minutes = date.getMinutes();
+  const seconds = date.getSeconds();
+
+  const formattedDate = `${day} ${month}, ${year}`;
+  const formattedTime = `${hours.toString().padStart(2, "0")}:${minutes
+    .toString()
+    .padStart(2, "0")}:${seconds.toString().padStart(2, "0")}`;
+
+  return `${formattedDate} - ${formattedTime}`;
+}
+
+export function getPipelinesTriggerTime(
+  pipelines: PipelineTriggerCount[]
+): string[] {
+  const triggerDates: string[] = [];
+
+  pipelines.forEach((pipeline) => {
+    pipeline.counts.forEach((count) => {
+      triggerDates.push(count.trigger_time);
+    });
+  });
+
+  // triggers.forEach((trigger) => {
+  //   if (!triggerTime.includes(formatDateTime(trigger.trigger_time))) {
+  //     triggerTime.push(formatDateTime(trigger.trigger_time));
+  //   }
+  // });
+
+  const uniqueDates = Array.from(new Set(triggerDates)); // Convert Set to an array
+  uniqueDates.sort((a, b) => new Date(a).getTime() - new Date(b).getTime()); // Sort dates in ascending order
+  // const formattedDates = uniqueDates.map((date) => formatDateTime(date));
+  // return formattedDates;
+
+  return uniqueDates;
+}
+
+export function getPipelinesSeries(triggers: TriggerCount[]) {
+  return triggers.map((trigger) => {
+    return {
+      name: trigger.pipeline_id,
+      type: "line",
+      smooth: true,
+      data: trigger.counts.map((trigger) => trigger.count),
+    };
+  });
+}
+
+export function getStatus(status: string): ResourceState {
+  if (status === "completed") {
+    return "STATE_ACTIVE";
+  }
+  if (status === "errored") {
+    return "STATE_ERROR";
+  }
+  return "STATE_UNSPECIFIED";
+}
+
+export function getColor(
+  statusname: string
+): TagProps & { iconColor?: string; label?: string } {
+  switch (statusname) {
+    case "STATE_ERROR":
+    case "errored":
+      return {
+        variant: "lightRed",
+        iconColor: "semantic-error-default",
+        label: "Error",
+      };
+
+    case "STATE_ACTIVE":
+    case "completed":
+      return {
+        variant: "lightGreen",
+        iconColor: "semantic-success-default",
+        label: "Active",
+      };
+
+    case "STATE_INACTIVE":
+      return {
+        variant: "default",
+        label: "Inactive",
+      };
+
+    default:
+      return {
+        variant: "default",
+        label: "Unknown",
+      };
+  }
+}
```

Comment:
This function has lots of unexpected side-effect. It name is getColor, but what it gets is a object with variant and label and color? This will confuse other developers